packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/utils/ReflectionUtils.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative.sessionreplay.utils
+
+import java.lang.reflect.Field
+
+internal class ReflectionUtils {
+    internal fun getDeclaredField(instance: Any, fieldName: String): Any? {
```

Comment:
oh wow...are we going to use reflection ? How many times is this code going to be called ?

Reply:
Unfortunately yes, since we need the values from the React shadownode. It would be called multiple times for every view

Reply:
I would test the performances of these methods and see how they could affect the frame rate. These methods are being called on the UIThread and it will be problematic. I would rather have less quality of the replays than affected performance.